### PR TITLE
docs: Create ADR for conflicting contributions

### PR DESCRIPTION
**Pull request by dbartholomae**
See #13 for previous related discussions

### CODE REVIEW

File: decisions/ADR-04-conflicting-contributions.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+Sometimes, new functionality for this library is suggested that is already covered by other open source projects. Multiple of those projects might try to get incorporated into this one.
+
+## Options
+
+### Forgo adding existing solutions
+
+We could forgo adding existing solutions to our codebase. This would get around any kind of related conflict and the solutions would still be available.
+
+On the other hand, one of the goals of this library is to provide thoroughly testede solutions to make it easier for enterprise users to consume just one approved library.
```
Comment (mbonig) [inline decisions/ADR-04-conflicting-contributions.md:18]:
```suggestion
On the other hand, one of the goals of this library is to provide thoroughly tested solutions to make it easier for enterprise users to consume just one approved library.
```

File: decisions/ADR-04-conflicting-contributions.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Combine learnings from multiple solutions
```
Comment (mbonig) [inline decisions/ADR-04-conflicting-contributions.md:20]: Should this be a '###'?

File: decisions/ADR-04-conflicting-contributions.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+We could research alternatives and try to combine what we learn from those that are willing to contribute to create an improved solution.
+
+### Decide for a competing solutino
```
Comment (go-to-k) [inline decisions/ADR-04-conflicting-contributions.md:24]:
```suggestion
### Decide for a competing solution
```

**Review by go-to-k — CHANGES_REQUESTED** [review]

File: decisions/ADR-04-conflicting-contributions.md (hunk 2 above)
Comment (dbartholomae) [reply decisions/ADR-04-conflicting-contributions.md:20]: Yes, thanks!

```suggestion
### Combine learnings from multiple solutions
```

**Review by go-to-k — APPROVED** [review]

**Pull request merged**